core/domain/classroom_config_domain_test.py

```diff
@@ -0,0 +1,182 @@
+from core.domain import classroom_config_domain
+from core.tests import test_utils
+
+
+class ClassroomDomainTests(test_utils.GenericTestBase):
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.classroom = classroom_config_domain.Classroom(
+            'classroom_id', 'math', 'math',
+            'Curated math foundations course.',
+            'Start from the basics with our first topic.',
+            {
+                'topic_id_1': ['topic_id_2', 'topic_id_3'],
+                'topic_id_2': [],
+                'topic_id_3': []
+            }
+        )
+        self.classroom_dict: classroom_config_domain.ClassroomDict = {
+            'classroom_id': 'classroom_id',
+            'name': 'math',
+            'url_fragment': 'math',
+            'course_details': 'Curated math foundations course.',
+            'topic_list_intro': 'Start from the basics with our first topic.',
+            'topic_id_to_prerequisite_topic_ids': {
+                'topic_id_1': ['topic_id_2', 'topic_id_3'],
+                'topic_id_2': [],
+                'topic_id_3': []
+            }
+        }
+
+    def test_that_domain_object_is_created_correctly(self) -> None:
+        self.assertEqual(self.classroom.classroom_id, 'classroom_id')
+        self.assertEqual(self.classroom.name, 'math')
+        self.assertEqual(self.classroom.url_fragment, 'math')
+        self.assertEqual(
+            self.classroom.course_details,
+            'Curated math foundations course.'
+        )
+        self.assertEqual(
+            self.classroom.topic_list_intro,
+            'Start from the basics with our first topic.'
+        )
+        self.assertEqual(
+            self.classroom.topic_id_to_prerequisite_topic_ids,
+            {
+                'topic_id_1': ['topic_id_2', 'topic_id_3'],
+                'topic_id_2': [],
+                'topic_id_3': []
+            }
+        )
+        self.classroom.validate()
+
+    def test_from_dict_method(self) -> None:
+        classroom = classroom_config_domain.Classroom.from_dict(
+            self.classroom_dict)
+
+        self.assertEqual(classroom.classroom_id, 'classroom_id')
+        self.assertEqual(classroom.name, 'math')
+        self.assertEqual(classroom.url_fragment, 'math')
+        self.assertEqual(
+            classroom.course_details,
+            'Curated math foundations course.'
+        )
+        self.assertEqual(
+            classroom.topic_list_intro,
+            'Start from the basics with our first topic.'
+        )
+        self.assertEqual(
+            classroom.topic_id_to_prerequisite_topic_ids,
+            {
+                'topic_id_1': ['topic_id_2', 'topic_id_3'],
+                'topic_id_2': [],
+                'topic_id_3': []
+            }
+        )
+
+    def test_to_dict_method(self) -> None:
+        self.assertEqual(self.classroom.to_dict(), self.classroom_dict)
+
+    def test_incorrect_classroom_id_should_raise_exception(self) -> None:
```

Comment:
I think you want "invalid" instead of "incorrect", here and elsewhere where it applies. 

"Valid" means passes the model validity checks. "Correct" means "is actually the right value" (e.g. it's 2 rather than 3).

Reply:
Thank you for the suggestion, updated PTAL.